### PR TITLE
Use browser natives when bundled with webpack

### PR DESCRIPTION
**Pull request by southpolesteve**
I work on [a library ](https://github.com/Azure/azure-cosmos-js) that depends on node-fetch and abort-controller.  When you bundle my library (and by proxy this library) with webpack it will include polyfills and shims even in browsers that don't need them. Instead by specifying a browser field in package.json you can rely on the browser natives. 

This is basically identical to how node-fetch handles it https://github.com/bitinn/node-fetch/blob/master/browser.js

### CODE REVIEW

**Comment by mysticatea** [comment]
Hi. Thank you for this PR.

I'm not familiar with the `browser` field.
If people wanted to use `abort-controller` with a bundler, for browsers which don't have `AbortController`, how do they do? People need to do nothing as is?

**Comment by southpolesteve** [comment]
The `browser` field directs bundlers like webpack and browserify to a different module entry point when bundling. This is helpful if you want to include (or omit) code specific to browsers when your code targets both node and the browser. I'll elaborate on the use cases and what this PR changes:

In the current setup, importing this module includes a polyfill in all environments: node, IE, and modern browsers. This is nice because it just works out of the box for everyone. But the vast majority of browsers users (non-IE) are including an unnecessary polyfill. There is not a great way to opt out of this behavior that I know of. Especially if you are consuming a library (like the azure cosmos SDK) that depends on this library and targets both node and the browser. If you only want to target modern browsers you are stuck shipping polyfill you don't need.

Adding `browser` field and pointing it to builtins changes things. Node users will still get poly filled automatically. Browser users will not. This is the ideal experience for all consumers, except of course for IE. Applications targeting IE will have to polyfill manually probably be adding something like `import 'abort-controller/polyfill'` to their code. 

TLDR: This improves the experience for most browser users, but will require application developers targeting IE to specifically import the polyfill.

**Comment by mysticatea** [comment]
Thank you for the elaboration!

Sounds good to me, but we need semver-major to publish it.
I will merge this PR.
